app/src/services/api.ts

```diff
@@ -26,5 +29,28 @@ export const api = createApi({
       transformResponse: (response: { data: InventoryProgressResponse }) =>
         response.data,
     }),
+    addCity: builder.mutation<CityAttributes, {}>({
```

Comment:
Would be good to add tags to the city query and using `invalidatesTags` here to clear the cached city.